### PR TITLE
Change `CheckSyncedGitCommits` to `IsGitCommitSynced`

### PR DESCRIPTION
**Pull request by saquibmian**
We never passed more than one hash in.

### CODE REVIEW

File: private/buf/bufsync/syncer.go [hunk 1]
```diff
@@ -516,26 +516,25 @@ func (s *syncer) branchSyncableCommits(
...
 	modIdentity := moduleIdentity.IdentityString()
 	// check local cache first
 	if syncedModuleCommits, ok := s.modulesIdentitiesToCommitsSyncedCache[modIdentity]; ok {
-		if _, commitSynced := syncedModuleCommits[commitHash]; commitSynced {
+		if _, commitSynced := syncedModuleCommits[commitHash.Hex()]; commitSynced {
 			return true, nil
 		}
 	}
 	// not in the cache, request BSR check
-	syncedModuleCommits, err := s.handler.CheckSyncedGitCommits(ctx, moduleIdentity, map[string]struct{}{commitHash: {}})
```
Comment (saquibmian) [inline private/buf/bufsync/syncer.go:528]: We only ever passed in one hash.

File: private/buf/cmd/buf/command/alpha/repo/reposync/sync_handler.go [hunk 2]
```diff
@@ -88,30 +87,22 @@ func (h *syncHandler) ResolveSyncPoint(ctx context.Context, module bufmoduleref.
...
-		syncedHash := label.LabelName.Name
-		if _, expected := commitHashes[syncedHash]; !expected {
-			return nil, fmt.Errorf("received unexpected synced hash %q, expected %v", syncedHash, commitHashes)
+			return false, nil
 		}
-		syncedHashes[syncedHash] = struct{}{}
+		return false, fmt.Errorf("get labels in namespace: %w", err)
 	}
-	return syncedHashes, nil
+	return len(res.Msg.Labels) == 1, nil
```
Comment (unmultimedio) [inline private/buf/cmd/buf/command/alpha/repo/reposync/sync_handler.go:105]: May be make sure that the returned label is actually the one requested?

File: private/buf/bufsync/bufsync.go [hunk 3]
```diff
@@ -167,14 +167,13 @@ type Handler interface {
 		branch string,
 	) (git.Hash, error)
 
-	// CheckSyncedGitCommits is invoked when syncing branches to know which commits hashes from a set
-	// are already synced inthe BSR. It expects to receive the commit hashes that are synced already. If
-	// an error is returned, sync will abort.
-	CheckSyncedGitCommits(
+	// IsGitCommitSynced is invoked when syncing branches to know if a Git commit is already synced.
+	// If an error is returned, sync will abort.
+	IsGitCommitSynced(
```
Comment (unmultimedio) [inline private/buf/bufsync/bufsync.go:172]: Yeah I remember when doing this I was imagining paging many git hashes in a single request, but with multi module directories that became increasingly difficult and right now we just do it one by one. I'm good with this change!

**Review by unmultimedio — APPROVED** [review]

File: private/buf/cmd/buf/command/alpha/repo/reposync/sync_handler.go (hunk 2 above)
Comment (saquibmian) [reply private/buf/cmd/buf/command/alpha/repo/reposync/sync_handler.go:105]: I thought about this, but this is going away soon anyway, to be replaced by `GetReferenceByName{ Ref: git.Hash() }`, so I didn't bother.

**Pull request merged**